pkg/controller/engine.go

```diff
@@ -281,7 +305,18 @@ func (c *namedController) Start(ctx context.Context) error {
 	return nil
 }
 
+// IndexField queues an index for addition to the cache on start.
```

Comment:
Would `QueueIndexField` or `IndexFieldOnStart` or something similar make this more self-descriptive?

What happens if this is called after the controller/cache is started? Nothing I guess? (Worth mentioning in doc string.)

Reply:
Good points.

Reply:
Actually, standby with this PR. The context passed to `GetInformer` and with that to `IndexField` is only used for waiting, not for the informer life-cycle. I think this was a misunderstanding.